ext/pdo_firebird/firebird_driver.c

```diff
@@ -1059,38 +1057,35 @@ static bool pdo_firebird_set_attribute(pdo_dbh_t *dbh, zend_long attr, zval *val
 
 		case PDO_FB_TRANSACTION_ISOLATION_LEVEL:
 			{
-				if (!pdo_get_long_param(&lval, val)) {
+				if (UNEXPECTED(!pdo_get_long_param(&lval, val))) {
 					return false;
 				}
 
-				if (H->in_manually_txn) {
+				if (UNEXPECTED(H->in_manually_txn)) {
 					pdo_raise_impl_error(dbh, NULL, "HY000", "Cannot change transaction isolation level while a transaction is already open");
 					return false;
 				}
 
 				/* ignore if the new value equals the old one */
 				if (H->txn_isolation_level != lval) {
-					if (lval == PDO_FB_READ_COMMITTED ||
-						lval == PDO_FB_REPEATABLE_READ ||
-						lval == PDO_FB_SERIALIZABLE
-					) {
-						/*
```

Comment:
This comment might still be valuable, what do you think ?

Reply:
Ah, indeed! If I decide to switch like this, I will leave a comment.

But I was thinking it might be better to make it an enum. I'll give it some more thought.

Either way, you're right, comments should be left.